File: python/source/interactive_books/domain/prompt_message.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from interactive_books.domain.tool import ToolInvocation


@dataclass(frozen=True)
class PromptMessage:
    role: str
    content: str
    tool_use_id: str | None = None
    tool_invocations: list[ToolInvocation] = field(default_factory=list)

File: swift/InteractiveBooks/Sources/InteractiveBooksCore/Domain/ValueObjects/PromptMessage.swift
public struct PromptMessage: @unchecked Sendable, Equatable {
    public let role: String
    public let content: String
    public let toolUseId: String?
    public let toolInvocations: [ToolInvocation]

    public init(
        role: String,
        content: String,
        toolUseId: String? = nil,
        toolInvocations: [ToolInvocation] = []
    ) {
        self.role = role
        self.content = content
        self.toolUseId = toolUseId
        self.toolInvocations = toolInvocations
    }
}
